### PR TITLE
Fix some problem under Windows OS development enviroment

### PR DESCRIPTION
**Pull request by Rokixy**
1. language.ts will be changed in every commit when commiting with a 'zh-cn' OS language setting. Fix it by specifying the locale 'en' when sorting languages with localeCompare().

2. The parameter of glob.sync() must in posix path form. In Win32 system the path separator '\\' considered as a escape char by node-glob, Fix it by replace '\\' with '/'.

3. build-api with child_process.execFile() will cause "Error: spawn yarn ENOENT" in Win32. Replace it with child_process.exec().
### Change Type

- [ ] `patch` — Bug fix
- [ ] `minor` — New feature
- [ ] `major` — Breaking change
- [ ] `dependencies` — Changes to package dependencies[^1]
- [ ] `documentation` — Changes to the documentation only[^2]
- [ ] `tests` — Changes to any test code only[^2]
- [x] `internal` — Any other changes that don't affect the published package[^2]
- [ ] I don't know

[^1]: publishes a `patch` release, for devDependencies use `internal`
[^2]: will not publish a new version

### Release Notes

- stablize language.ts when running under different OS language.
- add isWin32() and posixPath() to format the parameter of glob.sync().
- use child_process.exec() instead of child_process.execFile() for win32 platform.

### CODE REVIEW

File: scripts/lib/exec.ts [hunk 1]
```diff
@@ -27,50 +27,29 @@ export async function exec(
 ): Promise<string> {
 	nicelog(`> $ ${command} ${args.join(' ')} (in ${pwd}))`)
 	return new Promise((resolve, reject) => {
-		const data: string[] = []
-
-		const childProcess = execFile(
-			command,
-			args.filter((arg): arg is string => !!arg),
+		const childProcess = execute(
+			`${command} ${args.filter((arg): arg is string => !!arg).join(' ')}`,
```
Comment (SomeHats) [inline scripts/lib/exec.ts:31]: `execFile` was used here because it sanitizes the arguments passed to it. Can we restore sanitization here?

File: scripts/lib/exec.ts [hunk 2]
```diff
@@ -27,50 +27,29 @@ export async function exec(
...
 			{ cwd: pwd, env: { ...process.env, ...env } },
-			(err) => {
-				if (err) reject(err)
-				else resolve(data.join(''))
+			(err, stdout, stderr) => {
+				if (err) {
+					reject(err)
+				} else {
+					const combinedOutput = `${stdout}${stderr}`
+					resolve(combinedOutput.trim())
```
Comment (SomeHats) [inline scripts/lib/exec.ts:38]: We want `stdout` and `stderr` to be interleaved rather than one after the other. Why this change?

File: scripts/lib/exec.ts [hunk 3]
```diff
@@ -27,50 +27,29 @@ export async function exec(
...
+			childProcess.stdout.on('data', (data) => {
+				processStdoutLine(data)
+			})
+		}
+
+		if (childProcess.stderr) {
+			childProcess.stderr.on('data', (data) => {
+				processStderrLine(data)
+			})
+		}
```
Comment (SomeHats) [inline scripts/lib/exec.ts:53]: afaict, these `processLine` callbacks now might be dealing with larger chunks of data instead of single lines if the child process outputs either incomplete lines or multiple lines at once.

What's the intention behind removing the line processing?

**Review by SomeHats — CHANGES_REQUESTED** [review]
Thanks @Rokixy!

I'm a little hesitant for us to take all of these changes right now. In general, although we support the tldraw library on windows, we don't particularly aim to support developing this repo on win32. This repo does run within the [WSL](https://learn.microsoft.com/en-us/windows/wsl/) though which might help.

The language.ts stability thing seems generally useful - I would very happily land this on its own.

The other windows-specific changes i'm a little unsure about as they're likely to break in future as we don't develop for windows and don't have any tests. If you wanted to accompany this with a windows CI action that tests some of these scripts then that would be cool!

For the `exec` changes specifically, I think there are some regressions in the functionality of that helper that this introduces. I'd want to see those resolved before landing this. One option we could consider there is switching from plain `child_process` to https://github.com/sindresorhus/execa.

Sorry to be so resistant to this change! I just want to make sure we can support anything we do here going forward. In general, I would definitely suggest just working with this repo under the WSL instead.

**Comment by Rokixy** [comment]
Thank you for your review @SomeHats .

Everything works fine in WSL. I'm not sure about windows-specific changes and `exec` changes is needed. So I revert  those two commits.

**Review by SomeHats — APPROVED** [review]
Thanks! This is a great change
